tests/core/resource/test_record.py

```diff
@@ -347,6 +347,114 @@ def test_returns_changed_fields_if_behaviour_is_light(self):
         self.assertDictEqual(result, {'unread': True, 'position': 0})
 
 
+class JsonPatchTest(BaseTest):
+    def setUp(self):
+        super(JsonPatchTest, self).setUp()
+        self.stored = self.model.create_record({})
+        self.resource.record_id = self.stored['id']
+        self.resource.request.json = {'data': {'a': 'aaa', 'b': ['bb', 'bbb'], 'd': []}}
+        schema = ResourceSchema()
+        self.resource.mapping = schema
+        self.result = self.resource.patch()['data']
+        header = self.resource.request.headers
+        header['Content-Type'] = 'application/json-patch+json'
+        self.resource._is_json_patch = True
+
+    def test_json_patch_add(self):
+        self.resource.request.json = [
+            {'op': 'add', 'path': '/data/c', 'value': 'ccc'},
+            {'op': 'add', 'path': '/data/b/1', 'value': 'bbb'},
+            {'op': 'add', 'path': '/data/b/-', 'value': 'bbbb'},
+        ]
+        result = self.resource.patch()['data']
+        self.assertEqual(result['c'], 'ccc')
+        self.assertEqual(result['b'][2], 'bbb')
```

Comment:
I think this test is confusing because you're inserting between elements 0 and 1, but you're inserting the same value as element 1. I think this test should use `result['b'][1]` and its value should be different than `bbb`. Also, you might put add an assertion about the `len` here (since you have them in other tests too).